### ISSUE TITLE
Doc for torch.utils.data.distributed.DistributedSampler is confusing

### ISSUE THREAD

**Issue by bjourne** [pytorch/pytorch#40043]
In the documentation for this class there is the warning:

"In distributed mode, calling the set_epoch method is needed to make shuffling work; each process will use the same random seed otherwise."

After the warning follows an example. But the example doesn't call set_epoch. Thus it is not clear what the warning is trying to convey.

**Comment by SsnL** [comment]
fixed on master https://pytorch.org/docs/master/data.html#torch.utils.data.distributed.DistributedSampler
